### ISSUE TITLE
🐞 Test flaky on sdk/elixir

### ISSUE THREAD

**Issue by wingyplus** [dagger/dagger#5642]
### What is the issue?

Found on:

* https://github.com/dagger/dagger/actions/runs/5868523844/job/15912901044?pr=5628
* https://github.com/dagger/dagger/actions/runs/5877596164/job/15939249914
* https://github.com/dagger/dagger/actions/runs/5877596164/job/15939953049

Some tests spend to much time, causing test failure due to a timeout of over 60 seconds. Some tests got session timeout.

### SDK version

Elixir SDK main

### OS version

macOS, linux

**Comment by wingyplus** [comment]
The root cause

```
%Jason.DecodeError{position: 0, token: nil, data: "\e[35m1:\e[0m starting session \e[90m[0.26s]\e[0m\n\e[35m1:\e[0m connect \e[32mDONE\e[
0m\n{\"port\":45191,\"session_token\":\"8f5b735f-3071-489b-84f9-6d8671a27d12\"}\n"}
```

**Comment by gerhard** [comment]
The problem is that:
1. The SDK tries to connect via the dev CLI to the dev Engine and times out after 10 seconds
2. The SDK proceeds to download the latest stable CLI, which then fails with the above error

I think this step 2 is a mistake. I've attempted a fix here https://github.com/dagger/dagger/pull/5640 (see the diff snippet).  The problem remains: the SDK keeps timing out when connecting to the Engine. See these comments / questions in this order:
- https://github.com/dagger/dagger/pull/5628#issuecomment-1679175396
- https://github.com/dagger/dagger/pull/5628#issuecomment-1679378257
- https://github.com/dagger/dagger/pull/5628#issuecomment-1679670292 (I have not seen these failures in the Elixir SDK before this PR)
- https://github.com/dagger/dagger/pull/5640#issue-1853321922

When the connect timeout gets bumped to `30s` or `60`, the test times out (each test has a default `60s` timeout).

The question remains: why does the CLI (invoked from the SDK) not finish initialising the session after `10s`?

cc @vito

**Comment by wingyplus** [comment]
@gerhard I found that somehow when running inside dagger, such as internal/mage. Elixir Port doesn't separate `\n` as a newline. That cause session token cannot retrieve and causing session timeout, then fallback to download cli mode. I'm preparing to submit PR in a few minutes.

**Comment by gerhard** [comment]
Nice find! Btw, we are in the process of releasing `v0.8.3`. I propose that we skip publishing Elixir SDK `v0.8.3`, and instead get this fix in, then release `v0.8.4` next week, or the week after.

**Comment by wingyplus** [comment]
I'm fine with that. It would not be good if we publishing it while still having this issue. :)
